src/mesh/index_derivs.cxx

```diff
@@ -58,6 +58,7 @@ STAGGER Mesh::getStagger(const CELL_LOC inloc, const CELL_LOC outloc,
 STAGGER Mesh::getStagger(const CELL_LOC vloc, MAYBE_UNUSED(const CELL_LOC inloc),
                          const CELL_LOC outloc, const CELL_LOC allowedStaggerLoc) const {
   TRACE("Mesh::getStagger -- four arguments");
+  ASSERT1(inloc == outloc);
```

Comment:
I'm not sure this is needed/right -- `vloc==inloc` in the line below so if `outloc==inloc` this says everything has to be at exactly the same location but I thought we were slightly more flexible than this.

Reply:
We are more flexible: `vloc==inloc` is only one possibility in the line below. `vloc` is also allowed to be staggered relative to `inloc` and `outloc`, as long as one of `vloc` or `inloc` is CELL_CENTRE. The `vloc==inloc` case lets us also do unstaggered but not-`CELL_CENTRE` cases, e.g. `vloc=inloc=outloc=CELL_XLOW` for a y-derivative.